sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java

```diff
@@ -45,11 +52,16 @@ public DefaultLogger(Class<?> clazz) {
      * class name passes in.
      */
     public DefaultLogger(String className) {
+        String classPath;
         try {
-            this.classPath = Class.forName(className).getCanonicalName();
+            classPath = Class.forName(className).getCanonicalName();
         } catch (ClassNotFoundException e) {
-            this.classPath = className;
+            classPath = className;
         }
+        this.classPath = classPath;
+        this.configuredLogLevel =
+            LogLevel.fromString(Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_LOG_LEVEL))
+                .getLogLevel();
```

Comment:
`configuredLogLevel` is only ever set here in the constructor. I wonder if it is worth doing a micro-optimisation here to determine which log levels are valid and pre-compute the boolean conditions here, to simplify the `isTraceEnabled`, etc methods?

Reply:
Yeah, I had initially done it that way to have booleans for each of the log levels but the performance improvements were not noticeable by doing that way vs the current approach. 

Moreover, `is*Enabled` methods in DefaultLogger are much faster than their logback counterparts (557664530 ops/sec vs 501866764 ops/sec). So, didn't want to micro-optimize. 

However, if you think evaluating it once in the constructor will improve overall readability, I am fine with making this change.